system/jlib/jfile.hpp

```diff
@@ -468,11 +468,44 @@ extern jlib_decl void removeIFileCreateHook(IRemoteFileCreateHook *);
 
 extern jlib_decl IFile * createIFile(const RemoteFilename & filename);
 
-// Hook mechanism for accessing files inside containers (eg zipfiles)
+enum StorageType
```

Comment:
Minor: I'm not sure that this enum is very useful (it makes it less extensible).  Simpler would be to return the string, and have the hook do a string compare.